### ISSUE TITLE
counteq, countne, sqL2dist, L2dist, L1dist, L1infdist, gkldiv, meanad, maxad, msd, rmsd, and psnr may return incorrect results with offset indices

### ISSUE THREAD

**Issue by yurivish** [JuliaStats/StatsBase.jl#638]
The [counteq](https://juliastats.org/StatsBase.jl/stable/deviation/#StatsBase.counteq) function is meant to count the number of indices at which the elements of the arrays a and b are equal. However, there are situations in which this function can nondeterministically return incorrect results due to out-of-bounds memory accesses.

Here is an example with two arrays with identical indices and identical values:

```julia
julia> using StatsBase, OffsetArrays

julia> [
               let
                       a = OffsetArray(collect(1:11), -5:5)
                       b = copy(a)
                       counteq(a, b)
               end
               for _ in 1:10
       ]
10-element Array{Int64,1}:
 8
 8
 7
 7
 6
 7
 7
 6
 6
 7
 ```

The bug is caused by an incorrect use of [@inbounds](https://docs.julialang.org/en/v1/base/base/#Base.@inbounds).

I noticed that same bug exists in all of the other mathematical functions defined in the [same file](https://github.com/JuliaStats/StatsBase.jl/blob/320411b96c0ef0c5b0006169768b13f2554afb00/src/deviation.jl).

**Comment by nalimilan** [comment]
Yes these should use `for i in eachindex(a, b)`.

**Comment by ararslan** [comment]
Fixed by #867
